src/static/js/modules/util/utilities-type-checkers.js

```diff
@@ -0,0 +1,150 @@
+/* ==========================================================================
+   Javascript Type Checkers
+
+   Various utility functions to check Javascript types and primitives.
+   Copied from:
+   https://github.com/angular/angular.js/blob/master/src/Angular.js.
+   ========================================================================== */
+
+'use strict';
+
+/**
+ * @name isUndefined
+ * @kind function
+ *
+ * @description
+ * Determines if a reference is undefined.
+ *
+ * @param {*} value Reference to check.
+ * @returns {boolean} True if `value` is undefined.
+ */
+function isUndefined( value ) {
+  return typeof value === 'undefined';
+}
+
+
+/**
+ * @name isDefined
+ * @kind function
+ *
+ * @description
+ * Determines if a reference is defined.
+ *
+ * @param {*} value Reference to check.
+ * @returns {boolean} True if `value` is defined.
+ */
+function isDefined( value ) {
+  return typeof value !== 'undefined';
+}
+
+
+/**
+ * @name isObject
+ * @kind function
+ *
+ * @description
+ * Determines if a reference is an `Object`.
+ * Unlike `typeof` in JavaScript, `null`s are not
+ * considered to be objects. Note that JavaScript arrays are objects.
+ *
+ * @param {*} value Reference to check.
+ * @returns {boolean} True if `value` is an `Object` but not `null`.
+ */
+function isObject( value ) {
+  // http://jsperf.com/isobject4
+  return value !== null && typeof value === 'object';
+}
+
+
+/**
+ * @name isString
+ * @kind function
+ *
+ * @description
+ * Determines if a reference is a `String`.
+ *
+ * @param {*} value Reference to check.
+ * @returns {boolean} True if `value` is a `String`.
+ */
+function isString( value ) {
+  return typeof value === 'string';
+}
+
+
+/**
+ * @name isNumber
+ * @kind function
+ *
+ * @description
+ * Determines if a reference is a `Number`.
+ *
+ * This includes the "special" numbers `NaN`, `+Infinity` and `-Infinity`.
+ *
+ * If you wish to exclude these then you can use the native
+ * [`isFinite'](https://developer.mozilla.org/en-US/docs/Web/JavaScript/
+ *                      Reference/Global_Objects/isFinite)
+ * method.
+ *
+ * @param {*} value Reference to check.
+ * @returns {boolean} True if `value` is a `Number`.
+ */
+function isNumber( value ) {
+  return typeof value === 'number';
```

Comment:
This is generally going to be fine, but note this won't catch numbers created through the JS Number wrapper class, which it probably should. E.g. `typeCheckers.isNumber(new Number(1))` would return false.